processing/src/main/java/org/apache/druid/segment/ColumnProcessorFactory.java

```diff
@@ -76,6 +77,8 @@ public interface ColumnProcessorFactory<T>
    */
   T makeLongProcessor(BaseLongColumnValueSelector selector);
 
+  T makeArrayProcessor(BaseObjectColumnValueSelector<?> selector, ColumnCapabilities columnCapabilities);
```

Comment:
I think i was just matching the vector engine version, where all of them accept a capabilities and i find to be a nicer interface than the non-vectorized processor factory. Capabilities is more rich than just TypeSignature when we have access to it (and since these come from selector factories we do), so maybe the better thing to do here is make it @Nullable, since a null capabilities here means that the type was unknown

Reply:
I made this comment because I saw a bunch of extra work being done to new up a DefaultCapabilitiesImpl only to set the parameter and nothing else (under the assumption that it's going to be used by these methods that only care about the type and don't care about other things).  If you want to keep it as Capabilities, then I think you need to look at all `new DefaultCapabilitiesImpl` calls in this code and make 100% certain that they are not lying.